File: javascript/darts/darts.js
const OUTSIDE_TARGET_POINTS = 0
const OUTER_CIRCLE_BOUNDARY = 10
const OUTER_CIRCLE_POINTS = 1
const MIDDLE_CIRCLE_BOUNDARY = 5
const MIDDLE_CIRCLE_POINTS = 5
const INNER_CIRCLE_BOUNDARY = 1
const INNER_CIRCLE_POINTS = 10
const CALCULATE_POINT = (x, y) => Math.sqrt(x**2 + y**2)

export const score = (x, y) => {
  const point = CALCULATE_POINT(x, y)

  if (point > OUTER_CIRCLE_BOUNDARY) {
    return OUTSIDE_TARGET_POINTS
  }

  if (point > MIDDLE_CIRCLE_BOUNDARY) {
    return OUTER_CIRCLE_POINTS
  }

  if (point > INNER_CIRCLE_BOUNDARY) {
    return MIDDLE_CIRCLE_POINTS
  }

  return INNER_CIRCLE_POINTS
}

File: ruby/darts/darts.rb
class Darts
  OUTSIDE_TARGET_POINTS = 0
  private_constant :OUTSIDE_TARGET_POINTS
  OUTER_CIRCLE_BOUNDARY = 10
  private_constant :OUTER_CIRCLE_BOUNDARY
  OUTER_CIRCLE_POINTS = 1
  private_constant :OUTER_CIRCLE_POINTS
  MIDDLE_CIRCLE_BOUNDARY = 5
  private_constant :MIDDLE_CIRCLE_BOUNDARY
  MIDDLE_CIRCLE_POINTS = 5
  private_constant :MIDDLE_CIRCLE_POINTS
  INNER_CIRCLE_BOUNDARY = 1
  private_constant :INNER_CIRCLE_BOUNDARY
  INNER_CIRCLE_POINTS = 10
  private_constant :INNER_CIRCLE_POINTS

  def initialize(x_coord, y_coord)
    @point = Math.hypot(x_coord, y_coord)
  end

  def score
    if point > OUTER_CIRCLE_BOUNDARY
      OUTSIDE_TARGET_POINTS
    elsif point > MIDDLE_CIRCLE_BOUNDARY
      OUTER_CIRCLE_POINTS
    elsif point > INNER_CIRCLE_BOUNDARY
      MIDDLE_CIRCLE_POINTS
    else
      INNER_CIRCLE_POINTS
    end
  end

  private

  attr_reader :point
end
